pkg/apis/jenkins/v1alpha2/jenkins_types.go

```diff
@@ -214,6 +218,15 @@ type Plugin struct {
 	Version string `json:"version"`
 }
 
+// Plugin defines Jenkins agent
+type Agent struct {
+	// Name is the name of Jenkins Agent
+	Name string `json:"name"`
+	// Image to be used for the Jenkins Agent
+	Image string `json:"image"`
+	// ServiceAccountName to be used on the Jenins Agent
+	ServiceAccountName string `json:"serviceAccountName"`
+}
```

Comment:
- [ ] the doc comment needs to be more clear, and needs to start with `Agent`
- [ ] we need a different structure, at least something like something like
```
Replicas Integer `json:"replicas,omitempty"`
Template []PodTemplateSpec `json:"template,omitempty"`
```
Probably some more and/or different structs (I'm going from the docs here)